src/node_type.rs

```diff
@@ -663,8 +659,22 @@ impl InternalNode {
         unreachable!("Impossible to get here without returning even at the lowest level.")
     }
 
+    fn test_get_only_child(width: u8, range_existence_bitmap: u16, range_leaf_bitmap: u16) -> bool {
+        width == 1 || (range_existence_bitmap.count_ones() == 1 && range_leaf_bitmap != 0)
+    }
+
+    fn test_get_child(
```

Comment:
nit: how about `has_child`. Also, this seems like it would work well as a standard (non-associated) helper.